tools/src/main/java/com/nedap/archie/archetypevalidator/validations/NodeIdValidation.java

```diff
@@ -29,13 +30,15 @@ protected void beginValidation() {
 
     @Override
     public void validate(CObject cObject) {
+        int archetypeSpecialisationDepth = archetype.specializationDepth();
+
         if(cObject.getNodeId() == null) {
             //every object must have a node id
             addMessageWithPath(ErrorType.VCOID, cObject.getPath(),
                     I18n.t("C_OBJECT with RM type {0} must have a node id", cObject.getRmTypeName()));
         }
-        else if(!CPrimitiveObject.PRIMITIVE_NODE_ID_VALUE.equals(cObject.getNodeId()) && nodeIds.containsKey(cObject.getNodeId())) {
-            //every node id in a single archetype must be unique or a primitive object
+        else if(!CPrimitiveObject.PRIMITIVE_NODE_ID_VALUE.equals(cObject.getNodeId()) && archetypeSpecialisationDepth == AOMUtils.getSpecializationDepthFromCode(cObject.getNodeId()) && nodeIds.containsKey(cObject.getNodeId())) {
+            //every node id in a single archetype must be unique or a primitive object, or a occurrences matches {0} because sometimes that's the only way
```

Comment:
> or a occurrences matches {0} because sometimes that's the only way

When does this happen? Could you give an example?

Reply:
parent:
```
CLUSTER[id5] matches {
     ELEMENT[id1]
     ELEMENT[id2]
}
```

Child:
```
CLUSTER[id5.1] matches {
     ELEMENT[id1] occurrences matches {0}
}
CLUSTER[id5.2] matches {
     ELEMENT[id1] occurrences matches {0}
}
```

Totally valid ADL and the only way to do this, but not without the beforementioned change. Discussed with SEC group, the spec will be updated to match this.